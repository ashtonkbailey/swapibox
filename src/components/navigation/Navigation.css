.nav-container {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
}

.nav-button {
    width: 160px;
    height: 70px;
    font-size: 20px;
    border-radius: 20px;
    font-family: 'Poller One', cursive;
    background-color: rgb(244, 217, 68);
    box-shadow: 10px 5px 5px rgba(255, 255, 255, 0.25), inset 15px 5px 3px rgba(255, 255, 255, 0.25);
    z-index: 4;
}

.nav-button:hover {
    background-color:  rgb(220, 210, 158);
    cursor: pointer;
    transition:  .3s ease-in-out;
    font-size: 22px;
}

.reset-btn {
  position: absolute;
  top: 5%;
  left: 5%;
  background: none;
  border: none;
  color: #F3E355;
  font-family: Roboto;
  font-size: 18px;
  height: 40px;
  text-align: justify;
  text-shadow: 1px 1px 1px yellow;
  font-family: 'Poller One', cursive;
  font-weight: 200;
}

.reset-btn:hover {
  
}