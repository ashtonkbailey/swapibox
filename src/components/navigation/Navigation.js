import React from 'react';
import PropTypes from 'prop-types';

import './Navigation.css';

const Navigation = displayChosenContent => (
  <div className="nav-container">
    <button
      type="button"
      className="nav-button people"
      name="people"
      onClick={displayChosenContent}
    >
      people
    </button>
    <button
      type="button"
      className="nav-button planets"
      name="planets"
      onClick={displayChosenContent}
    >
      planets
    </button>
    <button
      type="button"
      className="nav-button vehicles"
      name="vehicles"
      onClick={displayChosenContent}
    >
      vehicles
    </button>
  </div>
);

Navigation.propTypes = {
  displayChosenContent: PropTypes.func.isRequired,
  content: PropTypes.array
}

export default Navigation;
