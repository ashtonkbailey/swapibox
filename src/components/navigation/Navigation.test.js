import React from 'react';
import { shallow } from 'enzyme';

import Navigation from './Navigation';

describe('nagivation', () => {
  let wrapper;
  let mockDisplayContent;

  beforeEach(() => {
    mockDisplayContent = jest.fn();
    wrapper = shallow(<Navigation displayChosenContent={mockDisplayContent} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display chosen content on button click', () => {
    const button = wrapper.find('.people');
    button.simulate('click');
    expect(mockDisplayContent).toBeCalled();
  });
});
