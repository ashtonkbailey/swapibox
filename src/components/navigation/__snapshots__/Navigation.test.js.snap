// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nagivation should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    displayChosenContent={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="reset-btn"
        >
           home 
        </button>,
        <button
          className="nav-button people"
          name="people"
          onClick={[MockFunction]}
          type="button"
        >
          people
        </button>,
        <button
          className="nav-button planets"
          name="planets"
          onClick={[MockFunction]}
          type="button"
        >
          planets
        </button>,
        <button
          className="nav-button vehicles"
          name="vehicles"
          onClick={[MockFunction]}
          type="button"
        >
          vehicles
        </button>,
      ],
      "className": "nav-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " home ",
          "className": "reset-btn",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": " home ",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people",
          "className": "nav-button people",
          "name": "people",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "people",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "planets",
          "className": "nav-button planets",
          "name": "planets",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "vehicles",
          "className": "nav-button vehicles",
          "name": "vehicles",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "vehicles",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="reset-btn"
          >
             home 
          </button>,
          <button
            className="nav-button people"
            name="people"
            onClick={[MockFunction]}
            type="button"
          >
            people
          </button>,
          <button
            className="nav-button planets"
            name="planets"
            onClick={[MockFunction]}
            type="button"
          >
            planets
          </button>,
          <button
            className="nav-button vehicles"
            name="vehicles"
            onClick={[MockFunction]}
            type="button"
          >
            vehicles
          </button>,
        ],
        "className": "nav-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " home ",
            "className": "reset-btn",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": " home ",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people",
            "className": "nav-button people",
            "name": "people",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": "people",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "planets",
            "className": "nav-button planets",
            "name": "planets",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": "planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "vehicles",
            "className": "nav-button vehicles",
            "name": "vehicles",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": "vehicles",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
